### ISSUE TITLE
Trial/reporting feature

### ISSUE THREAD

**Issue by Camology** [resilience4j/resilience4j#1406]
Resilience4j version: 1.3.1

Java version: 1.8.0_281

Not so much an issue but more of a question, we are implementing circuitbreakers on a project and things are going great locally but before we use them in production were were wondering if there was a simple way to enable only the events of a circuitbreaker flipping and not the actual fallbacks/circuitbreaker flipping. We want to see what the events look like in a production environment under normal use cases without actually flipping breakers so we can dial in our config file. 

So basically, we want the event publisher to fire off state changes, log those with our internal logging, but not actually flip a breaker/fall back to a fallback method and just continue as normal. Is there something we are missing that would already enable this, or a simple idea to do this other than just re-implementing our function in the fallback method?

**Comment by RobWin** [comment]
You could set the state of all CircuitBreaker instances to METRICS_ONLY at startup.

**Comment by Camology** [comment]
Perfect I think that is exactly what we are looking for. Will report back if it does and close the issue. Thanks @RobWin  !

**Comment by RobWin** [comment]
The problem that you can set the state via configuration properties. You need some code which iterates over all instances and sets the state before you accept requests.

**Comment by Camology** [comment]
We have a Circuitbreaker kind of factory that can do that at startup so shouldn't be an issue. Thanks though!

**Comment by Camology** [comment]
@RobWin  I forgot to update I apologize, but this worked very well for us thank you. Including an example in case anyone comes across this in the future:


```
public ResilienceConfiguration(CircuitBreakerRegistry circuitBreakerRegistry,
         @Value("${spring.application.name}") String appName,
         @Value("${customConfig.metrics-only:true}") boolean metricsOnly) {
                circuitBreakerRegistry.getAllCircuitBreakers()
                        .forEach(circuitBreaker -> {
                            if (metricsOnly) {
                                circuitBreaker.transitionToMetricsOnlyState();
                            }
                            //someOtherStuffWasHere
    }
```
